app/plugins/newgrounds.ts

```diff
@@ -62,18 +76,62 @@ export class NewgroundsPlugin extends BaseSitePlugin {
     }
 
     async checkFileDownload(): Promise<I.Media> {
+        // old comment:
         // Only check file downloads for art; other media has to be treated specially.
-        return this.getArtMedia();
+
+        // new comment: check 'em all!
+
+        let downloadable = this.getArtMedia();
+        if(downloadable) {
+            return downloadable;
+        }
+
+        downloadable = this.getVideoMedia();
+        if(downloadable) {
+            return downloadable;
+        }
+
+        downloadable = this.getAudioMedia();
+        if(downloadable) {
+            return downloadable;
+        }
+
+        return null;
     }
 
     getArtMedia(): I.Media {
-        const pageImgAnchor: HTMLAnchorElement = querySelector("#portal_item_view");
-        if (!pageImgAnchor) {
+        logger.log("newgrounds: art");
+
+        // There can be slightly different markup for "small" and "large"
+        // images.  Small images (the cutoff is roughly 500x600) do not
+        // have click-to-embiggen, but larger ones do.  The file type
+        // (JPG, PNG, GIF, WebP) doesn't seem to matter - only the pixel
+        // size.  Check both possibilities.
+
+        // large: ... div.pod_body div.image a#portal_item_view img
+        // small: ... div.pod_body div.image img
+        const largeImgElt: HTMLElement = querySelector("#portal_item_view img");
+        const smallImgElt: HTMLElement = querySelector("div.image img");
+
+        logger.log("newgrounds: large image, small image ", largeImgElt, smallImgElt);
+
+        let urlWithBuster = null;
+
+        if(largeImgElt) {
+            urlWithBuster = largeImgElt.getAttribute('src');
+        }
+        else if(smallImgElt) {
+            urlWithBuster = smallImgElt.getAttribute('src');
+        }
+        else {
+            logger.log("newgrounds: couldn't find any image, bailing out");
             return null;
         }
-        const pageImg = pageImgAnchor.querySelector('img');
-        const fullUrl = pageImgAnchor.href;
-        const previewUrl = pageImg.src;
+
+        // Trim ?f123... cache-buster off of end of URL
+        const fullUrl = urlWithBuster.split('?')[0];
+        const previewUrl = fullUrl;
```

Comment:
The intent with preview URL is that it should point to the smaller in-page image if one is available, so we should probably use it with smallImgElt's src attribute.

Reply:
I think I looked at this for one or two images and the "small" and "large" links pointed to exactly the same image at Newgrounds - in other words, Newgrounds isn't serving a thumbnail that fits in the space, and then a different, larger, image when you click-to-embiggen.  I *think* it's just serving one image, and either using a script or letting the browser scale it into the available space.

In some cases, only one of largeImgElt and smallImgElt ends up being populated, so we have to use that URL for both the main and preview URLs.

I will look at this further and report.

Reply:
Ah, good to know. Whichever works in that case.